pkg/load/configAgent.go

```diff
@@ -112,7 +122,9 @@ func (a *agent) loadFilenameToConfig() error {
 	a.configs = configs
 	a.generation++
 	a.lock.Unlock()
-	log.WithField("duration", time.Since(startTime)).Info("Configs reloaded")
+	duration := time.Since(startTime)
+	(*a.reloadMetrics).Observe(float64(duration.Nanoseconds() / 1e6))
```

Comment:
Why do you need to dereference the pointer?

Reply:
That method only works on the non-dereferenced value. Otherwise: `type *prometheus.Histogram has no field or method Observe`

Reply:
I'm not sure this will do what you want. Why not define the metrics you want in this package as `var` and do the registration as normal? Instead of trying to pass them through

Reply:
It now declares the `reloadMetrics` in the `configAgent`'s file. It still passes in the `errorRate` metric, since both the `main.go` and `configAgent.go` use it (and it's a pointer, so it doesn't have the same problem)